src/builder/edit_guild.rs

```diff
@@ -68,8 +68,8 @@ impl EditGuild {
     /// ```
     ///
     /// [`utils::read_image`]: crate::utils::read_image
-    pub fn icon(&mut self, icon: Option<&str>) -> &mut Self {
-        self.0.insert("icon", icon.map_or_else(|| NULL, |x| Value::from(x.to_string())));
+    pub fn icon(&mut self, icon: Option<impl Into<String>>) -> &mut Self {
```

Comment:
Taking `Option<impl Into<String>>` requires specifying a concrete type via turbofish whenever `None` is passed in (e.g. `None::<&str>`), so this doesn't work.

Reply:
hmm, is there a way I can specify a default trait bound to fix this?

Reply:
As it stands, the user can accomplish the same thing by just saying `s.map(|val| val.to_string()).as_deref()` to get an `Option<&str>` from a concrete `Option<impl Into<String>>`.

Reply:
The closure could be replaced by `Into::into`, however that doesn't fix the `None` problem... if I can't set a default bound and `None::<&str>` is unacceptable I'd prefer it to be `Option<String>` (as then the preference for owned strings is clearly shown)

Reply:
I'd rather just set the parameter to `Option<String>` and be done with it. It would keep things simple and would be up to the user to either pass ownership or not.